test/modules/aws-ec2-integration.ts

```diff
@@ -1,9 +1,49 @@
+import { EC2 } from '@aws-sdk/client-ec2';
 import config from '../../src/config';
 import * as iasql from '../../src/services/iasql'
 import { getPrefix, runQuery, runInstall, runUninstall, runApply, finish, execComposeUp, execComposeDown, runSync, } from '../helpers'
 
 const dbAlias = 'ec2test';
-const region = process.env.AWS_REGION;
+const region = process.env.AWS_REGION ?? '';
+const accessKeyId = process.env.AWS_ACCESS_KEY_ID ?? '';
+const secretAccessKey = process.env.AWS_SECRET_ACCESS_KEY ?? '';
+const ec2client = new EC2({
+  credentials: {
+    accessKeyId,
+    secretAccessKey
+  },
+  region
+});
+
+const getAvailabilityZones = async () => {
+  return await ec2client.describeAvailabilityZones({
+    Filters: [
+      {
+        Name: 'region-name',
+        Values: [region,],
+      },
+    ],
+  })
+}
+
+const getInstanceTypeOffering = async (availabilityZones: string[]) => {
+  return await ec2client.describeInstanceTypeOfferings({
+    LocationType: 'availability-zone',
+    Filters: [
+      {
+        Name: 'location',
+        Values: availabilityZones,
+      },
+      {
+        Name: 'instance-type',
+        Values: ['t2.micro', 't3.micro'],
+      }
+    ],
+  });
+}
```

Comment:
out of the scope of this PR, but this makes me think it might make sense to introduce an instance type table with fk to the availability zone table in `aws_account` I mention in https://github.com/iasql/iasql-engine/pull/941/files#r915165321?

Reply:
Hmmm instance type table should exist in aws ec2 module, I think. AvailabilityZone table not sure is aws_account or if it should be part of  vpc?

Reply:
yes, an `instance type` table would be in ec2. I suggested `availability_zone` in `aws_account` seems you mentioned different accounts have different availability zones